app/controllers/article_controller.rb

```diff
@@ -208,4 +214,13 @@ def eds_authenticated_user?
   def has_search_parameters?
     params[:q].present? || params[:f].present?
   end
+
+  def extract_fulltext_link(document, type)
+    links = document.fetch('eds_fulltext_links', [])
+    links.each do |link|
+      next if link['url'].blank? || link['url'] == 'detail'
+      return link['url'] if link['type'] == type.to_s
+    end
+    raise ArgumentError, "Missing #{type} fulltext link in document #{document.id}"
```

Comment:
@jvine what should we do in the failure scenario here?  As-written, I think we would give the user the generic rails 500 page (unless I'm missing something @drh-stanford).

Another option would be to redirect back to the url the user came from w/ a flash error message explaining what happened (and we can notify Honeybadger about the issue so we can track instances of it).

Thoughts?